packages/util/src/bytes.ts

```diff
@@ -23,7 +18,13 @@ export const bytesToUnprefixedHex = _bytesToUnprefixedHex
 /**
  * @deprecated
  */
-export const unprefixedHexToBytes = _unprefixedHexToBytes
+export const unprefixedHexToBytes = (inp: string) => {
+  if (inp.slice(0, 2) === '0x') {
+    throw new Error('hex string is prefixed with 0x, should be unprefixed')
+  } else {
```

Comment:
We can‘t do this check here since an unprefixed hex string can very well start with 0x. 😅 🤯

That‘s one of the reasons why we take prefixed as preferred: one can have a positive check that the format is correct.

Reply:
Actually that also means that you cannot use the method from ethereum-cryptography, now that I am thinking about it, right? So what you can do is always to add 0x and then use our own prefixed method I would think.

Reply:
Sorry, I thought we decided we wanted to explicitly check here if it was unprefixed? But, I can change, and then call into our own method (so we do not import `hexToBytes` from `ethereum-cryptography` anymore)

Reply:
Ah wait, maybe let's give this another round. Or would the Ox-in-string case only count for the utf8ToBytes version? 🤔

Reply:
>We can‘t do this check here since an unprefixed hex string can very well start with 0x

How is this possible, or have I misunderstood something along the way?  Hex doesn't include `x`, right, so unprefixed would explicitly mean no `0x`, right?

Reply:
Sorry, I think I am actually off here all the way down.

Have to some extend mixed this up with arbitrary (UTF-8) strings (fee very free to challenge me more on these kind of things, I stated this explicitly in the whole reasoning for the prefixed-hex work).

Anyhow: I think the outcome is (gladly enough, puh) still desired to have this unified since this reduces the risk of errrors when working in an inconsistent way with prefixed and unprefixed hex strings.

Ok, will update the branch here and merge.